src/main/src/test/java/org/geoserver/ows/StylePublisherTest.java

```diff
@@ -0,0 +1,145 @@
+import static org.junit.Assert.assertNotNull;
+
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.mock.web.MockServletContext;
+
+/**
+ *
+ * @author Ian Schneider <ischneider@boundlessgeo.com>
+ */
+public class StylePublisherTest extends GeoServerSystemTestSupport {
+
+    static StylePublisher publisher;
+
+    static List<String[]> paths = new ArrayList<String[]>();
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+
+        Catalog catalog = getCatalog();
+        publisher = new StylePublisher(catalog);
+        publisher.setServletContext(new MockServletContext());
+        GeoServerResourceLoader resourceLoader = getResourceLoader();
+
+        // add style - global
+        resourceLoader.copyFromClassPath("org/geoserver/ows/smileyface.png",
+                "styles/smileyface.png");
+        paths.add(new String[] { "styles", "smileyface.png" });
+
+        // add style - workspaced
+        resourceLoader.copyFromClassPath("org/geoserver/ows/house.svg",
+                "workspaces/cite/styles/house.svg");
+        paths.add(new String[] { "styles", "cite", "house.svg" });
+
+        // add style - workspaced style with global image
+        paths.add(new String[] { "styles", "cite", "smileyface.png" });
+
+        // testing over-riding global image with workspaced image
+        // global --> happyface
+        // ws'ed --> grass
+        resourceLoader.copyFromClassPath("org/geoserver/ows/smileyface.png", "styles/override.png");
+        resourceLoader.copyFromClassPath("org/geoserver/ows/grass_fill.png",
+                "workspaces/cite/styles/override.png");
+    }
+
+    private MockHttpServletResponse request(String[] path, String modifiedSince) throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.setContextPath("/geoserver");
+        request.setMethod("GET");
+        StringBuilder b = new StringBuilder("/geoserver");
+        for (int i = 0; i < path.length; i++) {
+            b.append('/').append(path[i]);
+        }
+        String uri = URLEncoder.encode(b.toString(), "UTF-8");
+        request.setRequestURI(uri);
+        if (modifiedSince != null) {
+            request.addHeader("If-Modified-Since", modifiedSince);
+        }
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        publisher.handleRequest(request, response);
+        return response;
+    }
+
+    @Test
+    public void testEncoding() throws Exception {
```

Comment:
this test appears to be testing the contents (via response.getContentAsByteArra(), rather then the "encoding" - consider renaming?